src/docfx/config/ConfigLoader.cs

```diff
@@ -86,6 +88,34 @@ string GetBranch()
             }
         }
 
+        private static void UpdateFileScope(JObject configObject)
```

Comment:
instead do if here, keep the raw config unchanged and merge it at `BuildScope.CreateGlob` method